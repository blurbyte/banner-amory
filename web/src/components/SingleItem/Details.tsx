// a bit more details about an item

import * as React from 'react';
import styled from 'styled-components';

import { Item } from '../../types/Item';
import Property from './Property';

const Wrapper = styled.div`
  margin-left: 3rem;
`;

const Description = styled.p`
  margin: 0;
  margin-bottom: 2rem;
  font-size: 1.8rem;
  line-height: 1.2;
  font-style: italic;
`;

type DetailsProps = Pick<
  Item,
  | 'description'
  | 'gamePart'
  | 'rank'
  | 'price'
  | 'acquirementMarketplace'
  | 'acquirementQuest'
  | 'acquirementGodstone'
  | 'acquirementHero'
  | 'trivia'
>;

class Details extends React.Component<DetailsProps> {
  render() {
    const {
      description,
      gamePart,
      rank,
      price,
      acquirementMarketplace,
      acquirementQuest,
      acquirementGodstone,
      acquirementHero,
      trivia
    } = this.props;
    return (
      <Wrapper>
        {description && <Description>{description}</Description>}
        {/* part */}
        <Property>
          <Property.Label>Part</Property.Label>
          <Property.Detail>{`The Banner Saga ${gamePart}`}</Property.Detail>
        </Property>
        {/* rank */}
        <Property>
          <Property.Label>Rank</Property.Label>
          <Property.Detail big>{rank}</Property.Detail>
        </Property>
        {/* price */}
        {price && (
          <Property>
            <Property.Label>Price</Property.Label>
            <Property.Detail big>{price}</Property.Detail>
          </Property>
        )}
        {/* acquirements */}
        <Property>
          <Property.Label>How to get</Property.Label>
          {acquirementMarketplace && <Property.Detail>Bought from market</Property.Detail>}
          {acquirementQuest && <Property.Detail>{acquirementQuest}</Property.Detail>}
          {acquirementGodstone && <Property.Detail>{`Found at ${acquirementGodstone} godstone`}</Property.Detail>}
          {acquirementHero && <Property.Detail>{`${acquirementHero} starts with it`}</Property.Detail>}
        </Property>
        {/* trivia */}
        {trivia && (
          <Property>
            <Property.Label>Trivia</Property.Label>
            <Property.Detail>{trivia}</Property.Detail>
          </Property>
        )}
      </Wrapper>
    );
  }
}

export default Details;
