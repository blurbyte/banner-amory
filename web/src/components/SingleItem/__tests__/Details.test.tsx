import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Details from '../Details';
import { testItem } from './mocks';

test('renders correctly', () => {
  const component = renderer.create(<Details {...testItem} />);
  expect(component.toJSON()).toMatchSnapshot();
});
