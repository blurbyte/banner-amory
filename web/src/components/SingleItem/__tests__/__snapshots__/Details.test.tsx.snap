// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c2 {
  margin: 1rem 0;
}

.c3 {
  font-size: 1.2rem;
  color: rgba(61,38,218,0.5);
  text-transform: uppercase;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: inherit;
}

.c4 {
  color: #3d26da;
  line-height: inherit;
  font-size: 1.4rem;
  font-weight: inherit;
  margin: 0;
}

.c5 {
  color: #3d26da;
  line-height: 1;
  font-size: 1.8rem;
  font-weight: 600;
  margin: 0;
}

.c6 {
  font-weight: 600;
}

.c0 {
  margin-left: 2rem;
}

.c1 {
  margin: 0;
  margin-bottom: 2rem;
  font-size: 1.8rem;
  line-height: 1.2;
  font-style: italic;
}

<div
  className="c0"
>
  <p
    className="c1"
  >
    Lovingly made by a hunter's wife to keep her daughter safe from harm.
  </p>
  <dl
    className="c2"
  >
    <dt
      className="c3"
    >
      Part
    </dt>
    <dd
      className="c4"
    >
      The Banner Saga 1
    </dd>
  </dl>
  <dl
    className="c2"
  >
    <dt
      className="c3"
    >
      Rank
    </dt>
    <dd
      className="c5"
    >
      1
    </dd>
  </dl>
  <dl
    className="c2"
  >
    <dt
      className="c3"
    >
      How to get
    </dt>
    <dd
      className="c4"
    >
      <strong
        className="c6"
      >
        Alette
      </strong>
       starts with it
    </dd>
  </dl>
</div>
`;
